raiden/tests/unit/test_tokennetwork.py

```diff
@@ -264,3 +266,116 @@ def test_multiple_channel_states(
 
     assert len(ids_to_channels) == 2
     assert channel_state.identifier in ids_to_channels
+
+
+def test_routing_updates(
+        token_network_state,
+        our_address,
+):
+    open_block_number = 10
+    pseudo_random_generator = random.Random()
+    pkey1, address1 = factories.make_privkey_address()
+    pkey2, address2 = factories.make_privkey_address()
+    pkey3, address3 = factories.make_privkey_address()
+
+    amount = 30
+    our_balance = amount + 50
+    channel_state = factories.make_channel(
+        our_balance=our_balance,
+        our_address=our_address,
+        partner_balance=our_balance,
+        partner_address=address1,
+    )
+    payment_network_identifier = factories.make_payment_network_identifier()
+
+    # create a new channel as participant, check graph update
+    channel_new_state_change = ContractReceiveChannelNew(
+        transaction_from=our_address,
+        token_network_identifier=token_network_state.address,
+        channel_state=channel_state,
+    )
+
+    channel_new_iteration1 = token_network.state_transition(
+        payment_network_identifier=payment_network_identifier,
+        token_network_state=token_network_state,
+        state_change=channel_new_state_change,
+        pseudo_random_generator=pseudo_random_generator,
+        block_number=open_block_number,
+    )
+
+    graph_state = token_network_state.network_graph
```

Comment:
please, don't rely on the mutation of the original object, use `channel_new_iteration1.new_state` here.